src/hotspot/share/services/mallocTracker.hpp

```diff
@@ -292,16 +292,12 @@ class MallocMemorySummary : AllStatic {
 class MallocHeader {
 
   NOT_LP64(uint32_t _alt_canary);
-  size_t _size;
-  uint16_t _bucket_idx;
-  uint16_t _pos_idx;
-  uint8_t _flags;
-  uint8_t _unused;
+  const size_t _size;
+  const uint32_t _mst_marker;
```

Comment:
make mst_marker a struct? instead of opaque type.

Reply:
I played around a lot with different forms (struct, union) and in the end settled on an opaque uint32 since
- it would be passed by value and I would have to provide that structure in all kind places, I got include circularities
- I have a vague improvement in my head where we store the malloc site table entries not as individually malloced elements but in a (dynamically growing) array; that would mean we could address them by index without having to walk the bucket chains; and that index would be a simple number.